scaleway/resource_k8s_cluster.go

```diff
@@ -436,8 +464,12 @@ func resourceScalewayK8SClusterCreate(ctx context.Context, d *schema.ResourceDat
 
 	// Private network configuration
 
+	isKapsule := clusterType == "" || strings.HasPrefix(clusterType.(string), "kapsule")
 	if pnID, ok := d.GetOk("private_network_id"); ok {
-		req.PrivateNetworkID = scw.StringPtr(expandZonedID(pnID.(string)).ID)
+		if !isKapsule {
+			return diag.FromErr(fmt.Errorf("only Kapsule clusters support private networks"))
+		}
+		req.PrivateNetworkID = scw.StringPtr(expandRegionalID(pnID.(string)).ID)
 	}
```

Comment:
Can this check be achieved in `CustomizeDiff` ?
You can also return a full diagnostic to improve error output
```go
diag.Diagnostic{
	Severity:      diag.Error,
	Summary:       "Private networks are not supported for current cluster type",
	Detail:        "Cluster type must be kapsule to use a private network",
	AttributePath: cty.GetAttrPath("private_network_id"),
}
```